src/ore/src/tracing/optimizer.rs

```diff
@@ -0,0 +1,313 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
```

Comment:
I don't think `ore` is the right place for all this, it should probably be in some other crate.